src/leafletwidget/widget.ts

```diff
@@ -80,28 +77,51 @@ class MapWidget extends Widget {
    */
   protected onUpdateRequest(msg: Message): void {
     this.title.text = this._context.path.split('/').pop();
-    let cm = this._context.contentsModel;
-    if (cm === null) {
+    if (!this.isAttached) {
       return;
     }
-    let content = JSON.parse(this._context.model.toString());
+    let contentString = this._context.model.toString();
+    if (contentString === this._geojsonString) {
```

Comment:
comparing json strings can fail on property order. The phosphor mono-repo contains a function for deep equality comparison of json objects. @blink1073 may have a temp copy of that lying around in this repo somewhere.